codegenerator/cli/templates/static/codegen/src/Address.res

```diff
@@ -15,10 +15,17 @@ module Evm = {
   // of generated code instead of adding it to the indexer project dependencies.
   let fromStringOrThrow = fromStringOrThrow
 
-  @module("viem")
-  external fromAddressOrThrow: t => t = "getAddress"
-  // Reassign since the function might be used in the handler code
-  // and we don't want to have a "viem" import there. It's needed to keep "viem" a dependency
-  // of generated code instead of adding it to the indexer project dependencies.
-  let fromAddressOrThrow = fromAddressOrThrow
+  exception InvalidAddress({address: string, message: string})
+  let sanitizeOrThrow = (address: t): t => {
+    switch address->toString->fromStringOrThrow {
+    | exception _ =>
+      raise(
+        InvalidAddress({
+          address: address->toString,
+          message: "Unable to parse address. Expected a 20-byte hex string starting with 0x.",
+        }),
```

Comment:
Not essential but It would be nice to have our own exception that's easy to track down. Should be given the logging context of a handler. And it should highlight the value passed with a communicative message to the user.

Reply:
I think it should be some sort of error handling mechanism where we can add context as in rust and then catch the error in the root of the indexer to log it and display to user.

Reply:
Currently, we are just throwing different exceptions, which is not the most solid way.

Reply:
Cool, I think we could build a nice "context" abstraction in a similar way. The most similar thing we have now is the logger we pass around which contains contextual values of the event hand/handler where the error occurred.

Reply:
Exactly what I was thinking about :)